File: Additional/7.Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        a = str(x)[::-1]
        if a[-1] == "-":
            a = "-" + a[:-1]
            while a[1] == "0":
                a = a[0] + a[2:]
            if int(a) > 2147483647 or int(a) < -2147483648:
                return 0
            return a
        while a[0] == "0" and len(a) > 1:
            a = a[1:]
        if int(a) > 2147483647 or int(a) < -2147483648:
            return 0
        return a

File: Additional/7.Reverse Integer.ts
function reverse(x: number): number {
  const maxNumber = 2147483647;
  const minNumber = -2147483648;
  const positiveX = Math.abs(x);
  const reversedX =  x < 0 
    ? parseInt([...positiveX.toString()].reverse().join("")) * -1
    : parseInt([...positiveX.toString()].reverse().join(""));
  const result = reversedX > maxNumber || reversedX < minNumber ? 0 : reversedX;
  return result;
};
